File: cpp/src/data_structure/reach/reach_line.cpp
#include "reachset/data_structure/reach/reach_line.hpp"

using namespace reach;

ReachLine::ReachLine(double p_lon_min, double p_lat_min, double p_lon_max, double p_lat_max) :
        p_lon_min(p_lon_min), p_lat_min(p_lat_min), p_lon_max(p_lon_max), p_lat_max(p_lat_max) {}

bool ReachLine::operator==(ReachLine const& other) const {
    if (this->p_lon_min == other.p_lon_min and this->p_lon_max == other.p_lon_max and
        this->p_lat_min == other.p_lat_min and this->p_lat_max == other.p_lat_max)
        return true;

    return false;
}

File: commonroad_reach/data_structure/reach/reach_line.py
from typing import Tuple

from shapely.geometry import LineString


class ReachLine:
    """
    Line segment used in geometric operations.

    A line segment is formed with two vertices.
    """

    def __init__(self, p_lon_min: float, p_lat_min: float, p_lon_max: float, p_lat_max: float):
        self.p_lon_min = p_lon_min
        self.p_lat_min = p_lat_min
        self.p_lon_max = p_lon_max
        self.p_lat_max = p_lat_max

        self._shapely_line_string = LineString([(p_lon_min, p_lat_min), (p_lon_max, p_lat_max)])

    def __repr__(self):
        return f"ReachLine({self.p_lon_min},{self.p_lon_max},{self.p_lat_min},{self.p_lat_max})"

    @property
    def shapely_object(self) -> LineString:
        return self._shapely_line_string

    @property
    def bounds(self) -> Tuple:
        return self._shapely_line_string.bounds
